### ISSUE TITLE
Cleaner syntax for Rules/Helpers files.

### ISSUE THREAD

**Issue by 3den** [yahoo/atomizer#133]
@thierryk @renatoi Since we will allow developers to create their custom rules we it is important to have a clean syntax for that is familiar, this suggestion is make the rules/helpers less verbose and more similar to css.

```js
// rules.js
module.exports = [
    ...    
    {
        name: 'Margin (all edges)',
        allowCustom: true,
        matcher: 'My',
        // this are the suffix aliases 
        arguments: [ 
          {
           a: auto',
           inh: 'inherit'
          },
          {
           default: '$0'
           a: auto',
           inh: 'inherit'
           }
       ],
        // The style should be very similar to the syntax of normal CSS
        style: {
            'margin-top': '$0',
            'margin-bottom': '$1'
        }
    },
    {
        // Inherit from the margin and just has to set what is different, 
        // all properties that are note defined here are reused from the parrent
        parent: 'margin'
        name: 'Margin top',
        prefix: 'Mt-',
        style: {
            'margin-top': '$suffix'
        }
    },
    ...
];
```

**Comment by 3den** [comment]
We can follow that same approach for helpers:

```js

Helpers should follow the same syntax:

```js
// helpers.js
module.exports = [
    ...    
    {
         name: 'LineClamp Helper',
         prefix: 'LineClamp-',
         allowCustom: true,
         style: {
              ' max-height': '$suffix' 
        },
        'style:static' 
              'display': 'inline-block',
              '*display': 'inline';
              'zoom': 1; 
         },
         'style:after' : {
            content: ".",
            'font-size': 0,
            visibility: hidden;
            display: inline-block;
            overflow: hidden;
            height: 0;
           width: 0;
        }
    }
];
```

The code above would generate the following css:

```css
// This css is static and will be added just once 
[class*=LineClamp-] {
   display: -webkit-box;
   -webkit-box-orient: vertical;
   overflow: hidden;
}
a[class*=LineClamp-] {
   display: inline-block;
   *display: inline;
   zoom: 1;
}
a[class*=LineClamp-]:after {
   content: ".";
   font-size: 0;
   visibility: hidden;
   display: inline-block;
   overflow: hidden;
   height: 0;
   width: 0;
}

// This css is dynamic and it just updates the custom value:
.LineClamp-72px {
   max-height: 72px;
}
```

**Comment by renatoi** [comment]
The LineClamp example wouldn't work like that because a helper should receive params and not suffixes. But I get the idea.

I believe it's a nice to have feature. I need to go through all rules and see how it will look like and if I would find any specific problem that we didn't find in these examples.

**Comment by 3den** [comment]
@renatoi in that case maybe it would be better to call it `matcher` instead of `prefix`, a matcher can be more flexible than prefix/suffix:

```js
// helpers.js
module.exports = [
    ...    
    {
         name: 'LineClamp Helper',
         matcher: 'LineClamp($min, $max)',
         allowCustom: true,
         style: {
              'min-height': '$min'
              'max-height': '$max' 
        },
        'style:static' 
              'display': 'inline-block',
              '*display': 'inline';
              'zoom': 1; 
         },
         'style:after' : {
              content: ".",
              'font-size': 0,
              visibility: 'hidden';
              display: 'inline-block';
              overflow: 'hidden';
              height: 0;
              width: 0;
        }
    }
];
```

**Comment by renatoi** [comment]
Thanks @3den, this issue has been addressed in PR #169.
